### ISSUE TITLE
@fluentui/react-button: missing accessibility attributes when `as` prop is used

### ISSUE THREAD

**Issue by layershifter** [microsoft/fluentui#13671]
### Environment Information

- **Package version(s)**: @fluentui/react-button@0.3.2

```tsx
<Button
  as="div"
  content={{
    children: "Foo"
  }}
/>
```

https://codesandbox.io/s/funny-moon-xde2j?file=/src/App.js

#### Actual behavior:

`role="button"`, keyboard handler for enter/space, and `tabIndex="0"` are missing.

#### Expected behavior:

`role="button"` and `tabIndex="0"` are pressed when an element is `div`.

### Priorities and help requested:

Are you willing to submit a PR to fix? No
Requested priority: Normal

**Comment by dzearing** [comment]
Condensing the other space keyboard bug into this one.

**Comment by dzearing** [comment]
https://codesandbox.io/s/objective-allen-hh7yi?file=/src/App.js

This is now applied.

```jsx
<div role="button" data-isfocusable="true" tabindex="0" class="ms-Button root-52">
  <span class="content-54">Foo</span>
</div>
```
